api/src/main/java/org/pmiops/workbench/audit/targetproperties/WorkspaceTargetProperty.kt

```diff
@@ -0,0 +1,48 @@
+package org.pmiops.workbench.audit.targetproperties
+
+import java.util.function.Function
+import org.pmiops.workbench.model.Workspace
+
+internal data class WorkspaceTargetPropertyNameValue(val propertyName: String, val value: String)
+
+// N.B. entries will rarely be referred to by name, but are accessed via values(). I.e. they are
+// not safe to remove, even if an IDE indicates otherwise.
+enum class WorkspaceTargetProperty private constructor(val propertyName: String, private val extractor: Function<Workspace, String>) {
+    NAME(propertyName = "name", extractor = Function<Workspace, String> { it.name }),
```

Comment:
are the named parameters necessary? This class is a little hard to read since there's a lot going on. I think cutting that out could help.